test/e2e/simple_vlan_and_dynamic_ip_test.go

```diff
@@ -0,0 +1,52 @@
+package e2e
+
+import (
+	"fmt"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("NodeNetworkState", func() {
+	Context("when vlan configured", func() {
+		var (
+			vlanId = "102"
+		)
+
+		BeforeEach(func() {
+			updateDesiredState(ifaceUpWithVlanUp(*firstSecondaryNic, vlanId))
+			waitForAvailableTestPolicy()
+		})
+		AfterEach(func() {
+			updateDesiredState(vlanAbsent(*firstSecondaryNic, vlanId))
+			resetDesiredStateForNodes()
+		})
+		It("should have the vlan interface configured", func() {
+			for _, node := range nodes {
+				interfacesNameForNodeEventually(node).Should(ContainElement(fmt.Sprintf(`%s.%s`, *firstSecondaryNic, vlanId)))
+				vlanForNodeInterfaceEventually(node, fmt.Sprintf(`%s.%s`, *firstSecondaryNic, vlanId)).Should(Equal(vlanId))
```

Comment:
Could we drop the previous check and keep only this. I know that the util functions don't make it any easier, but I think it would make more sense.